ENIGMAsystem/SHELL/Graphics_Systems/OpenGL1/GLtextures.cpp

```diff
@@ -76,32 +69,35 @@ namespace enigma
       //glTexParameteri(GL_TEXTURE_2D, GL_TEXTURE_MAX_LEVEL, 3);
       glGenerateMipmap(GL_TEXTURE_2D);
     }
-    TextureStruct* textureStruct = new TextureStruct(texture);
-    textureStruct->width = width;
-    textureStruct->height = height;
-    textureStruct->fullwidth = fullwidth;
-    textureStruct->fullheight = fullheight;
-    textureStructs.push_back(textureStruct);
 
     //texture must be constructed before unbinding the texture so that it can apply its initial sampler state
     glBindTexture(GL_TEXTURE_2D, 0);
 
-    return textureStructs.size()-1;
+    Texture* textureStruct = new Texture();
+    textureStruct->width = width;
+    textureStruct->height = height;
+    textureStruct->fullwidth = fullwidth;
+    textureStruct->fullheight = fullheight;
+    const int id = textures.size();
+    textures.push_back(textureStruct);
+    texture_peers.resize(textures.size());
+    texture_peers[id] = texture;
+    return id;
```

Comment:
@JoshDreamland I must mention that I am concerned it's rather fragile using a second vector for the peer objects. The main issue is that we have to remember to keep them at the same size and while this only required an extra line in just a few places, I fear it could quickly grow into a wild fire. What's sad is that #1125 would make this a non issue because we would probably design the generic resource container to solve this specific problem. Here are some things we could do:
* Implement #1125 and use a generic resource container for the peers.
* Use a hash map instead of a second vector, but Josh hates this since the integer ids are clearly small and would result in collisions
* Add a void pointer to the generic texture object to associate the peer object instead of a second vector
* Use a template parameter for the generic texture object to associate the peer, but this would require keeping the definition of the generic texture objects vector in each graphics system

Reply:
Ok: f7788eea59c4a4938c4716d876443f7f403dcc11

We went with using virtual inheritance to avoid the second vector. We feel this is less precarious than keeping two vectors at the same size and synchronized.